xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -347,8 +362,21 @@ public void updateBalancingState(ConnectivityState newState, SubchannelPicker ne
         currentChildState = newState;
         currentChildPicker = newPicker;
 
```

Comment:
You need update `currentChildPicker`. Also, if `newState` is not `READY`, no need to wrap.

Reply:
No I do not need to update `currentChildPicker` here, `currentChildPicker` can still be the unwrapped picker. The wrapped picker is passed to `updateChildState` method and be used as the childPicker in `WeightedChildPicker`. The interception happens only at the time when `pickSubchannel()` is executed. See [test](https://github.com/grpc/grpc-java/blob/a134c68c24c92f10716b00c6daf2009bd6e9bebb/xds/src/test/java/io/grpc/xds/LocalityStoreTest.java#L222).

Also we do not need to worry about only wrapping if `newState` is not `READY` as this wrapped picker will not be considered if the `childState` is not `READY` (checked in [`updateChildState`](https://github.com/grpc/grpc-java/blob/a134c68c24c92f10716b00c6daf2009bd6e9bebb/xds/src/main/java/io/grpc/xds/LocalityStore.java#L281)).

Reply:
You need `currentChildPicker` because it is used to update the `childPickers` list in `updateChildState()`. Your test did not cover the case with multiple localities.

Reply:
Thanks for catching it.